e2e_tests/tests/test_sdk.py

```diff
@@ -474,3 +474,32 @@ def test_rp_workspace_mapping() -> None:
     finally:
         for wid in workspace_ids:
             bindings.delete_DeleteWorkspace(session=sess, id=wid)
+
+
+@pytest.mark.e2e_cpu
+def test_experiment_w_template() -> None:
+    sess = api_utils.user_session()
+    detobj = client.Determined._from_session(sess)
+    # create template
+    tpl = bindings.v1Template(
+        name=api_utils.get_random_string(),
+        config=conf.load_config(conf.fixtures_path("templates/template.yaml")),
+        workspaceId=1,
+    )
+    tpl_resp = bindings.post_PostTemplate(sess, body=tpl, template_name=tpl.name)
+    # create experiment with template
+    with open(conf.fixtures_path("no_op/single-one-short-step.yaml")) as f:
+        config = util.yaml_safe_load(f)
+    config["hyperparameters"]["num_validation_metrics"] = 2
+    emptydir = tempfile.mkdtemp()
```

Comment:
I think the [pytest temp-dir fixture](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html) is better practice. I know you're just following an existing pattern in the file, and this recommendation is ok to ignore.

But I think having an example of the fixture in the codebase is better than consistency in this case. Maybe future editors will copy/paste the better example.

Bonus points (also unnecessary) if you think it's a good use of your time to port the existing `tempfile` to this pattern.

Reply:
Thanks. Continuous improvement. This makes our tests better.